_posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc

```diff
@@ -1810,37 +1993,159 @@ Set the variables for the `account-baseline-app` module in this environment in t
+  ##################################
+
+  # By granting access to the root ARN of the Security account in each of the roles below,
+  # we allow administrators to further delegate access to other IAM entities
+
+  # A role that allows administrator access to the account.
+  allow_full_access_from_other_account_arns = [local.security_account_root_arn]
+
+  # A role for developers to use to access services in the account.
+  # Access to services is managed using the dev_permitted_services input.
+  allow_dev_access_from_other_account_arns = [local.security_account_root_arn]
+
+  # Assuming the developers role will grant access to these services.
+  dev_permitted_services = [
+    "ec2",
+    "ecs",
+    "lambda",
+    "rds",
+    "elasticache",
+    "route53",
+  ]
+
+  # A role to allow users that can view and modify AWS account billing information.
+  allow_billing_access_from_other_account_arns = [local.security_account_root_arn]
+
+  # A role that allows read only access.
+  allow_read_only_access_from_other_account_arns = [local.security_account_root_arn]
+
+  # A role that allows access to support only.
+  allow_support_access_from_other_account_arns = [local.security_account_root_arn]
+
+  # A list of account root ARNs that should be able to assume the auto deploy role.
+  allow_auto_deploy_from_other_account_arns = [
+    # External CI/CD systems may use an IAM user in the security account to perform deployments.
+    local.security_account_root_arn,
+
+    # The shared account contains automation and infrastructure tools, such as CI/CD systems.
+    "arn:aws:iam::${local.accounts.shared}:root",
+  ]
+
+  # Assuming the auto-deploy role will grant access to these services.
+  auto_deploy_permissions = [
+    "iam:GetRole",
+    "iam:GetRolePolicy",
+  ]
+
+  # Configures the auto deploy max session duration to be 4 hours.
+  max_session_duration_machine_users = 14400
+
+  # Configures the max session duration for roles that humans use to be 8 hours.
+  max_session_duration_human_users = 28800
+
+  service_linked_roles = ["autoscaling.amazonaws.com"]
+  ##################################
+  # KMS grants
+  ##################################
 
-  # Specify the services the auto-deploy IAM role will have access to
-  auto_deploy_permissions = ["cloudwatch:*", "logs:*", "dynamodb:*", "ecr:*", "ecs:*", "iam:GetPolicy", "iam:GetPolicyVersion", "iam:ListEntitiesForPolicy", "eks:DescribeCluster", "route53:*", "s3:*", "autoscaling:*", "elasticloadbalancing:*", "iam:GetRole", "iam:GetRolePolicy", "iam:PassRole"]
+  # These grants allow the autoscaling service-linked role to access to the AMI encryption key so that it
+  # can launch instances from AMIs that were shared from the shared-services account.
+  kms_grant_regions = {
+    ami_encryption_key = local.aws_region
+  }
+  kms_grants = {
+    ami_encryption_key = {
+      kms_cmk_arn       = "arn:aws:kms:us-east-1:${local.accounts.shared}:alias/ami-encryption"
```

Comment:
NIT: I might remove the reference to `us-east-1` here, using `local.aws_region` or some such, instead.
Or maybe this whole arn should be a placeholder text.